### PR TITLE
render: add simple functions to render rectangles and ellipses

### PR DESCRIPTION
**Pull request by emersion**
This PR ports the simple renderer interface from #319.

Updates #558

### CODE REVIEW

File: include/wlr/render/wlr_renderer.h [hunk 1]
```diff
@@ -36,14 +36,24 @@ bool wlr_render_texture(struct wlr_renderer *r, struct wlr_texture *texture,
 bool wlr_render_texture_with_matrix(struct wlr_renderer *r,
 	struct wlr_texture *texture, const float matrix[static 9], float alpha);
 /**
- * Renders a solid quad in the specified color.
+ * Renders a solid rectangle in the specified color.
  */
-void wlr_render_colored_quad(struct wlr_renderer *r,
+void wlr_render_rect(struct wlr_renderer *r, const struct wlr_box *box,
```
Comment (SirCmpwn) [inline include/wlr/render/wlr_renderer.h]: I prefer the name quad, after all the matrix could deform it into a non-rectangular shape

Comment (emersion) [reply include/wlr/render/wlr_renderer.h]: Then we should rename `ellipse` to something else.

Comment (SirCmpwn) [reply include/wlr/render/wlr_renderer.h]: nah cause no matter how deformed it gets it'll still be an ellipse

Comment (emersion) [reply include/wlr/render/wlr_renderer.h]: Our ellipse is defined by a rectangle (and just a rectangle), so it's the same issue.

**Pull request merged**